docs/get-started/ide/phpstorm/debugging.md

```diff
@@ -0,0 +1,142 @@
+
+### Add IDE Key to your php.ini
+
+Edit php.ini Template inside MAMP, search for the xdebug section and add:
+
+```ini php.ini
+xdebug.idekey=PHPSTORM
+```
+
+![MAMP PHP INI Settings](./assets/php_ini_setup.png)
+
+Hint: Use `Ctrl` + `F` and search for "xdebug"
+
+:::note
+
+After you have changed some settings make sure to save & restart your MAMP Server.
+
+:::
+
+### Check PHP Configuration in your Joomla! Installation
+
+Check in the PHP Settings in your Joomla! Backend in `System` > `System Information` > `PHP
+Information` and make sure that the Section for Xdebug is visible:
+
+![Joomla! PHP Information](./assets/xdebug_top.jpg)
+
+### Install Xdebug Browser Extension
+
+Install a Xdebug Extension of your choice in your Browser. Search for Xdebug in your Browser's Extensions Dictionary
+and enable the Plugin - Check your Browser's Guide on how to install Plugins. Please note: An installation may require
+a Browser restart.
+
+### Setting up Environments in PhpStorm
+
+Mostly you will write your Code in its own Project Folder but when using Joomla! The project will be zipped / installed
+or deployed from inside your IDE into to your Joomla Dev Environment. Which means you cannot use debugger directly
+inside the Project where you write your extension code (because it is not the file which will be run by Joomla!
+normally).
+
+So beside your extension development project in PhpStorm open or create a new Project for Debugging inside your local
+Joomla! Installation. This Project should point to the Joomla! Main directory of your local Joomla! Installation. So in
+PhpStorm click on `File` > `New Project form Existing Files`and select your Joomla! Installation. PhpStorm may ask you
+if you would like to Trust this Folder, confirm that request.
+
+### Configure PhpStorm Debugger
+
+In this new Project select `Run` from the Main Menu and Select `Edit Configurations`.
+
+![Edit Configurations Menu](./assets/run_edit_configurations.png)
+
+This will open a new window where you can add configuration.
+
+![Empty remote debug](./assets/empty_run_debug_config.png)
+
+Click on `Add new` here you can then add a new `PHP Remote Debug` Configuration.
+
+![Select remote debug](./assets/select_php_remote_debug.png)
+
+Name the Configuration as you like but for example "MAMP" (default is Unnamed).
+Tick the option `Filter debug connection by IDE Key` and click on the Button with the three dots beside the Server
+Selection.
+
+![Select remote debug](./assets/start_configuration.png)
+
+This will open a new PopUp window to set up the server settings. Enter here your MAMP's Server Information based on your
+MAMP Settings.
+The name can be "MAMP Server" or whatever you like. By Default MAMP Server's are running on Port 8890 use the Port of
+your Server here. In this case here the Joomla! Installation is available on the URL `https://joomla5.local:8890` so the
+Host Name is `joomla5.local`. Set now the Hostname and Port based on your configuration.
```

Comment:
.local is a reserved tld top level domain which is used for mdns.

please use joomla5.example.com for documentation or .internal for real internal dns usage.

Reply:
right... used now "joomla5.localhost" because the guide implies local development with MAMP